src/network/core/os_abstraction.h

```diff
@@ -23,9 +23,30 @@
 #include <ws2tcpip.h>
 #include <windows.h>
 
-#define GET_LAST_ERROR() WSAGetLastError()
+/**
+ * Get the last error code from any of the OS's network functions.
+ * What it returns and when it is reset, is implementation defined.
+ * @return The last error code.
+ */
+#define NetworkGetLastError() WSAGetLastError()
 #undef EWOULDBLOCK
 #define EWOULDBLOCK WSAEWOULDBLOCK
+
+/**
+ * Return the string representation of the last error from the OS's network functions.
+ * @return The error message, potentially an empty string but never \c nullptr.
+ */
+static inline const char *NetworkGetLastErrorString()
```

Comment:
Shouldn't we make `error` a parameter, so we can delay looking up the string in some cases? See other remarks about places we now print numbers instead of strings.